src/main/java/seedu/address/logic/newcommands/CreateDeadlineCommand.java

```diff
@@ -0,0 +1,139 @@
+     * @throws NoSuchTaskException Exception thrown due to invalid Task.
+     */
+    @Override
+    public CommandResult execute(AbsolutePath currPath, Root root) throws CommandException,
+            InvalidPathException, UnsupportedPathOperationException, InvalidIdException, NoSuchTaskException {
+        requireAllNonNull(currPath, root);
+        absolutePath = currPath.resolve(path);
+        CommandResult returnStatement = null;
+        if (absolutePath.isStudentDirectory()) {
+            StudentOperation student = studentOperation(root, absolutePath);
+            if (student.getAllTasks()
+                    .stream()
+                    .filter(x -> x.getDescription().equals(this.deadline.getDescription())) != null) {
+                throw new CommandException(MESSAGE_DUPLICATE_DEADLINE_TASK);
+            }
+            student.addTask(deadline);
+            GroupOperation group = groupOperation(root, absolutePath);
+            StudentId studentId = absolutePath.getStudentId().get();
+            stu = group.getChild(studentId);
+            returnStatement = new CommandResult(String.format(MESSAGE_SUCCESS, stu.toString()));
+        } else if (absolutePath.isGroupDirectory()) {
+            GroupOperation group = groupOperation(root, absolutePath);
+            if (group.getAllTasks()
+                    .stream()
+                    .filter(x -> x.getDescription().equals(this.deadline.getDescription())) != null) {
+                throw new CommandException(MESSAGE_DUPLICATE_DEADLINE_TASK);
+            }
+            group.addTask(deadline);
+            RootOperation currRoot = rootOperation(root);
+            GroupId groupId = absolutePath.getGroupId().get();
+            grp = currRoot.getChild(groupId);
+            returnStatement = new CommandResult(String.format(MESSAGE_SUCCESS, grp.toString()));
+        } else {
+            throw new CommandException(MESSAGE_INCORRECT_DIRECTORY_ERROR);
+        }
+        return returnStatement;
+    }
+
+    /**
+     * Compares this {@code CreateDeadlineCommand} to another {@code CreateDeadlineCommand} to see if they are equal.
+     *
+     * @param other The other object to compare against this {@code CreateDeadlineCommand}.
+     * @return True if the object is same as {@code CreateDeadlineCommand} and false otherwise.
+     */
+    @Override
+    public boolean equals(Object other) {
+        if (other == this) {
+            return true;
+        }
+
+        // instanceof handles nulls
+        if (!(other instanceof CreateDeadlineCommand)) {
+            return false;
+        }
+
+        CreateDeadlineCommand otherCreateDeadlineCommand =
+                (CreateDeadlineCommand) other;
+
+        return deadline.getDescription().equals(otherCreateDeadlineCommand.deadline.getDescription())
+                && deadline.getDueBy().equals(otherCreateDeadlineCommand.deadline.getDueBy())
+                && this.absolutePath.equals(otherCreateDeadlineCommand.absolutePath);
```

Comment:
Will it be better to use the equals method of the object, instead of manually do the equality check. Some issues found in other command class equals method.
Reason: If there is changes in the class field, also need to do changes on all the equality check.